src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb

```diff
@@ -1765,6 +1765,32 @@ Namespace Microsoft.CodeAnalysis.Operations
                 Return New LazyInvalidOperation(children, _semanticModel, syntax, type, constantValue, isImplicit)
             End If
         End Function
+
+        Private Function CreateBoundReDimOperation(boundRedimStatement As BoundRedimStatement) As IReDimOperation
+            Dim preserve As Boolean = boundRedimStatement.Syntax.Kind = SyntaxKind.ReDimPreserveStatement
+            Dim clauses As Lazy(Of ImmutableArray(Of IReDimClauseOperation)) = New Lazy(Of ImmutableArray(Of IReDimClauseOperation))(
+                Function()
+                    Return boundRedimStatement.Clauses.SelectAsArray(Function(n)
+                                                                         Debug.Assert(preserve = n.Preserve)
+                                                                         Return CreateBoundReDimClauseOperation(n)
```

Comment:
Consider going through the standard Visit to cache the result.

Reply:
I retained consistency in this file by calling the strongly typed method if the bound child has a stronger type guarantee and the required operation result also needs a stronger type (`ImmutableArray(Of IReDimClauseOperation)` here). For example, see `CreateBoundInterpolatedStringExpressionOperation`.

---
In reply to: [224239881](https://github.com/dotnet/roslyn/pull/30217#discussion_r224239881) [](ancestors = 224239881)

Reply:
Unfortunately this file is not consistent, but there are far more instances of using `DirectCast(Create(` than the strongly typed version.

---
In reply to: [224250368](https://github.com/dotnet/roslyn/pull/30217#discussion_r224250368) [](ancestors = 224250368,224239881)

Reply:
Fixed.

---
In reply to: [224254595](https://github.com/dotnet/roslyn/pull/30217#discussion_r224254595) [](ancestors = 224254595,224250368,224239881)